Modules/Relationships/src/Relationships.ConsumerApi/Controllers/RelationshipsController.cs

```diff
@@ -195,6 +197,17 @@ public async Task<IActionResult> DecomposeRelationship([FromRoute] string id, Ca
         return Ok(response);
     }
 
+    [HttpGet("CanCreate")]
+    [ProducesResponseType(typeof(HttpResponseEnvelopeResult<CanEstablishRelationshipResponse>), StatusCodes.Status200OK)]
+    [ProducesError(StatusCodes.Status400BadRequest)]
+    public async Task<IActionResult> CanEstablishRelationship([FromQuery] IdentityAddress peer, CancellationToken cancellationToken)
+    {
+        if (peer == null!) throw new ValidationException(GenericApplicationErrors.Validation.InvalidPropertyValue(nameof(peer)));
```

Comment:
You can remove this after adding the Validator to the Mediatr Query

Reply:
Done